Bottom: 614520780a8b216704eef3e883b67cee0d45c7fc
Top:    df30abd419fec513be894c366e616f2fabacd739
Author: Johannes Rudolph <johannes_rudolph@gmx.de>
Date:   2009-08-16 12:26:00 +0200

ojf: handle ParseExceptions gracefully in the plugin


---

diff --git a/objectformatter/src/main/scala/net/virtualvoid/string/EnhancedStringsPlugin.scala b/objectformatter/src/main/scala/net/virtualvoid/string/EnhancedStringsPlugin.scala
index 3b14afc..2e1558f 100644
--- a/objectformatter/src/main/scala/net/virtualvoid/string/EnhancedStringsPlugin.scala
+++ b/objectformatter/src/main/scala/net/virtualvoid/string/EnhancedStringsPlugin.scala
@@ -73,8 +73,12 @@ class EnhancedStringsPlugin(val global: Global) extends Plugin {
 	     */
 	    def postTransform(tree: Tree): Tree = tree match {
 	      case This(qual) => {System.out.println(tree+":"+qual+qual.getClass+":"+qual.toString.length);tree}
-	      case Literal(Constant(str:String)) =>
+	      case Literal(Constant(str:String)) => 
+	        try {
 	          compile(EnhancedStringFormatParser.parse(str))
+	        } catch {
+	          case p:ParseException => error(p.getMessage);tree
+	        }
 	      case _ => tree
 	    }
